node/node_handler.go

```diff
@@ -189,27 +249,44 @@ func (node *Node) BroadcastCrossLink() {
 		"Construct and Broadcasting new crosslink to beacon chain groupID %s",
 		nodeconfig.NewGroupIDByShardID(shard.BeaconChainShardID),
 	)
-	headers := []*block.Header{}
-	lastLink, err := node.Beaconchain().ReadShardLastCrossLink(curBlock.ShardID())
+
+	for _, shardID := range []uint32{1, 2, 3} {
+		lastLink, err := node.Blockchain().ReadShardLastCrossLink(shardID)
```

Comment:
We probably don't need to send the whole crosslink data from beacon to shard, just need to send a message containing the number of the last continuous crosslink. Of course, the sender (leader or validators) can sign the message to prove the legitimacy of the data to other shards.